src/main/scala/org/allenai/pipeline/Pipeline.scala

```diff
@@ -178,10 +180,11 @@ trait Pipeline extends Logging {
   def runOnly(title: String, targetNames: Iterable[String]): Iterable[(String, Any)] = {
     val targets = getStepsByName(targetNames)
     val targetStepInfo = targets.map(_._2.stepInfo).toSet
-    val allDependencies = targets.flatMap { case (s, p) => Workflow.upstreamDependencies(p) }
+    val directDependencies = targets.flatMap { case (s, p) => p.stepInfo.dependencies.map(_._2) }
```

Comment:
Bug fix: runOnly will throw exception if upstream dependencies don't exist, but should only check direct upstream dependencies